homeassistant/components/airly/air_quality.py

```diff
@@ -0,0 +1,219 @@
+        return res
+
+    return _decorator
+
+
+class AirlyAirQuality(AirQualityEntity):
+    """Define an Airly air_quality."""
+
+    def __init__(self, airly, name):
+        """Initialize."""
+        self.airly = airly
+        self.data = airly.data
+        self._name = name
+        self._pm_2_5 = None
+        self._pm_10 = None
+        self._aqi = None
+        self._icon = "mdi:blur"
+        self._attrs = {}
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return self._icon
+
+    @property
+    @round_state
+    def air_quality_index(self):
+        """Return the air quality index."""
+        return self._aqi
+
+    @property
+    @round_state
+    def particulate_matter_2_5(self):
+        """Return the particulate matter 2.5 level."""
+        return self._pm_2_5
+
+    @property
+    @round_state
+    def particulate_matter_10(self):
+        """Return the particulate matter 10 level."""
+        return self._pm_10
+
+    @property
+    def attribution(self):
+        """Return the attribution."""
+        return ATTRIBUTION
+
+    @property
+    def state(self):
+        """Return the CAQI description."""
+        return self.data[ATTR_API_CAQI_DESCRIPTION]
+
+    @property
+    def unique_id(self):
+        """Return a unique_id for this entity."""
+        return f"{self.airly.latitude}-{self.airly.longitude}"
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return bool(self.airly.data)
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        self._attrs[LABEL_ADVICE] = self.data[ATTR_API_ADVICE]
+        self._attrs[LABEL_AQI_LEVEL] = self.data[ATTR_API_CAQI_LEVEL]
+        self._attrs[LABEL_PM_2_5_LIMIT] = self.data[ATTR_API_PM25_LIMIT]
+        self._attrs[LABEL_PM_2_5_PERCENT] = round(self.data[ATTR_API_PM25_PERCENT])
+        self._attrs[LABEL_PM_10_LIMIT] = self.data[ATTR_API_PM10_LIMIT]
+        self._attrs[LABEL_PM_10_PERCENT] = round(self.data[ATTR_API_PM10_PERCENT])
+        return self._attrs
+
+    async def async_update(self):
+        """Get the data from Airly."""
+        await self.airly.async_update()
+
+        self._pm_10 = self.data[ATTR_API_PM10]
+        self._pm_2_5 = self.data[ATTR_API_PM25]
+        self._aqi = self.data[ATTR_API_CAQI]
+
+
+class AirlyData:
+    """Define an object to hold sensor data."""
+
+    def __init__(self, session, api_key, latitude, longitude, language, **kwargs):
+        """Initialize."""
+        self.session = session
+        self.latitude = latitude
+        self.longitude = longitude
+        self.language = language
+        self.api_key = api_key
+        self.airly = Airly(self.api_key, self.session, language=self.language)
+        self.data = {}
+
+        self.async_update = Throttle(kwargs[CONF_SCAN_INTERVAL])(self._async_update)
```

Comment:
We can decorate `_async_update` method directly with the `Throttle` and use the `DEFAULT_SCAN_INTERVAL` constant directly. Then rename `_async_update` to `async_update`.